lib/cdo/aws/cloudfront.rb

```diff
@@ -0,0 +1,211 @@
+require_relative '../../../deployment'
```

Comment:
My take:  Manual testing before initial deployment is OK, but we will need
to add automated testing to verify that the CDN is behaving as expected
before we can consider the CDN work done.  This isn't an area where we want
to incur technical debt because it's critical for both scale and
correctness , HOC is coming up very soon, and because development doesn't
give us good coverage for what the CDN is doing.

Is there any way we can assertions about caching headers to our existing UI
tests that go through the CDN to see if the CDN is behaving as expected?

On Mon, Oct 5, 2015 at 9:19 AM, Will Jordan <notifications@github.com>
wrote:

> In lib/cdo/aws/cloudfront.rb
> <https://github.com/code-dot-org/code-dot-org/pull/4207#discussion_r41164020>
> :
>
> > @@ -0,0 +1,211 @@
> > +require_relative '../../../deployment'
>
> Added CloudFront unit tests to test_cloudfront.rb
> <https://github.com/code-dot-org/code-dot-org/blob/b781b19ee8812d56eaeaeb0306d784772571fdb1/shared/test/test_cloudfront.rb>,
> with the following note:
>
> These unit tests simply confirm that the #create_or_update method will pass
> properly-structured data to the AWS CloudFront Client library.
> Separate integration tests are required to guarantee that the live API
> endpoints will accept the values provided.
>
> I think manual testing of the CloudFront distribution created by this
> script should be fine for now, and we can look into adding integration
> tests later if possible.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/code-dot-org/code-dot-org/pull/4207/files#r41164020>.
>